adapters/cryptopunks2/index.js

```diff
@@ -0,0 +1,123 @@
+require("dotenv").config();
+
+const moment = require("moment");
+const BigNumber = require("bignumber.js");
+const Ethereum = require("../../sdk/EVMC");
+const axios = require("axios");
+const URL = "http://nft-sales-service.dappradar.com/open-source";
+const KEY = process.env.DAPPRADAR_API_KEY;
+const path = require("path");
+
+class CRYPTOPUNKS2 {
+
+    constructor() {
+        this.name = "cryptopunks2";
+        this.symbol = "PUNKS2";
+        this.token = "0xc02d332AbC7f9E755e2b1EB56f6aE21A7Da4B7AD";
+        this.protocol = "polygon";
+        this.block = 20346370;
+        this.contract = "0xc02d332AbC7f9E755e2b1EB56f6aE21A7Da4B7AD";
+        this.events = ["Transfer"],
+        this.pathToAbi = path.join(__dirname, "./abi.json");
+        this.range = 500;
+        this.chunkSize = 6;
+        this.sdk = new Ethereum(this);
+    }
+
+    run = async () => {
+        const s = await this.getSymbol();
+        this.sdk = this.loadSdk();
+        this.symbol = s;
+        await this.sdk.run();
+    };
+
+    loadSdk = () => {
+        return new Ethereum(this);
+    };
+    getSymbol = async () => {
+        return "PUNKS2";
+    };
+    getPrice = async timestamp => {
+        let date = new Date(timestamp * 1000);
+        let dateDisp = date.getDate() + "-" + date.getUTCMonth() + "-" + date.getFullYear();
+        const query = "https://api.coingecko.com/api/v3/coins/matic-network/history?date=" + dateDisp;
```

Comment:
use our pricing api please

Reply:
Your API is returning 0 response for MATIC on polygon...

token: 0x0000000000000000000000000000000000001010
protocol: polygon

Resposne:
{"protocol":"POLYGON","token":"0x0000000000000000000000000000000000001010","date":"2022-01-04 09:00:00","price":0,"decimals":0,"currency":"usd"}

How should we proceed?

Reply:
Hi @Arnaslt 

Your API is returning 0 response for MATIC on polygon...

token: 0x0000000000000000000000000000000000001010
protocol: polygon

Resposne:
{"protocol":"POLYGON","token":"0x0000000000000000000000000000000000001010","date":"2022-01-04 09:00:00","price":0,"decimals":0,"currency":"usd"}

How should we proceed?

Reply:
protocol should be "matic" , thats what cause the issue I believe

Reply:
@Arnaslt 
Thanks, yes - this works for the pricing API... we'll adjust and resubmit this part

Reply:
KFMWWHYDUD8BWZ8UQ7TFYSGHZZ5GX1G3D9